react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+let counter = 0;
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { isClicked: counter };
+  }
+
+  handleClick() {
+    this.setState({ isClicked: counter++ });
+  }
+
+  render() {
+    if (counter < 3) {
+      return <button onClick={this.handleClick}>Click!</button>;
+    }
+    if (counter >= 3 && counter < 6) {
+      return <button onClick={this.handleClick} className="button-one">Click!</button>;
+    }
+    if (counter >= 6 && counter < 9) {
+      return <button onClick={this.handleClick} className="button-two">Click!</button>;
+    }
+    if (counter >= 9 && counter < 12) {
+      return <button onClick={this.handleClick} className="button-three">Click!</button>;
+    }
+    if (counter >= 12 && counter < 15) {
+      return <button onClick={this.handleClick} className="button-four">Click!</button>;
+    }
+    if (counter >= 15 && counter < 18) {
+      return <button onClick={this.handleClick} className="button-five">Click!</button>;
+    }
+    if (counter >= 18) {
+      return <button onClick={this.handleClick} className="button-six">Click!</button>;
+    }
```

Comment:
This is a lot of repetition and would be hard to maintain, it is better to figure out the class then return just one button.
```JSX
let btnClass = '';

if (counter >= 3 && counter < 6) {
  btnClass = "button-one";
} else if (counter >= 6 && counter < 9) {
  btnClass = "button-two";
} else if (counter >= 9 && counter < 12) {
  btnClass = "button-three";
} else if (counter >= 12 && counter < 15) {
  btnClass = "button-four";
} else if (counter >= 15 && counter < 18) {
  btnClass = "button-five";
} else if (counter >= 18) {
  btnClass = "button-six";
}

return <button onClick={this.handleClick} className={btnClass}>Click!</button>;
```